src/testers.jl

```diff
@@ -183,16 +183,16 @@ All keyword arguments except for `fdm` and `fkwargs` are passed to `isapprox`.
 function frule_test(f, xẋs::Tuple{Any, Any}...; rtol=1e-9, atol=1e-9, fdm=_fdm, fkwargs=NamedTuple(), kwargs...)
     _ensure_not_running_on_functor(f, "frule_test")
     xs, ẋs = first.(xẋs), last.(xẋs)
-    Ω_ad, dΩ_ad = frule((NO_FIELDS, ẋs...), f, xs...; fkwargs...)
-    Ω = f(xs...; fkwargs...)
+    Ω_ad, dΩ_ad = frule((NO_FIELDS, deepcopy(ẋs)...), f, deepcopy(xs)...; deepcopy(fkwargs)...)
+    Ω = f(deepcopy(xs)...; deepcopy(fkwargs)...)
     # if equality check fails, check approximate equality
     # use collect so can do vector equality
     # TODO: add isapprox replacement that works for more types
     @test Ω_ad == Ω || isapprox(collect(Ω_ad), collect(Ω); rtol=rtol, atol=atol)
 
     ẋs_is_ignored = ẋs .== nothing
     # Correctness testing via finite differencing.
-    dΩ_fd = _make_jvp_call(fdm, (xs...) -> f(xs...; fkwargs...), xs, ẋs, ẋs_is_ignored)
+    dΩ_fd = _make_jvp_call(fdm, (xs...) -> f(deepcopy(xs)...; deepcopy(fkwargs)...), xs, ẋs, ẋs_is_ignored)
```

Comment:
Same here regarding mutation of `f`.

```suggestion
    dΩ_fd = _make_jvp_call(fdm, (xs...) -> deepcopy(f)(deepcopy(xs)...; deepcopy(fkwargs)...), xs, ẋs, ẋs_is_ignored)
```